cmd/operator/main.go

```diff
@@ -393,6 +393,46 @@ func run(fs *flag.FlagSet) int {
 		}
 	}
 
+	// If Prometheus Agent runs in DaemonSet mode, check if
+	// the operator has proper RBAC permissions on the Daemonset resource.
+	if cfg.Gates.Enabled(operator.PrometheusAgentDaemonSetFeature) {
+		daemonSetGroupVersion := schema.GroupVersion{
+			Group:   "apps",
+			Version: "v1",
+		}
+
+		daemonSetSupported, err := checkPrerequisites(
```

Comment:
we can call `k8sutil.IsAllowed()` only: the DaemonSet resource should always be supported since it's a Kubernetes core API.